walk/config.go

```diff
@@ -135,8 +135,12 @@ func (c *Configurer) loadBazelIgnore(repoRoot string, wc *walkConfig) error {
 			log.Printf("the .bazelignore exclusion pattern must not be a glob %s", ignore)
 			continue
 		}
-		// Ensure we remove trailing slashes or the exclude matching won't work correctly
-		wc.excludes = append(wc.excludes, strings.TrimSuffix(ignore, "/"))
+
+		// Ensure we remove leading "./" and trailing "/" or the exclude matching won't work correctly
+		ignore = strings.TrimPrefix(ignore, "./")
```

Comment:
Should we also strip this prefix if it is repeated?

Reply:
I think that's unlikely enough it's not worth the extra code 🤷

Reply:
Actually I'm quite wrong there, using `filepath.Clean` is simpler and I think more "correct" then the use of `TrimPrefix` we had before.

WDYT? See the latest and new test case with the ugly path in it.